### PR TITLE
AMP-44813 Android Java example

### PR DESCRIPTION
**Pull request by falconandy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,3 +15,6 @@ Example apps using the Ampli SDK
 
 ## Android Kotlin
 * [kotlin/android/AmpliApp](kotlin/android/AmpliApp)
+
+## Android Java
+* [java/android/AmpliApp](java/android/AmpliApp)
```
Comment (justin-fiedler) [inline README.md:20]: Same as comment on Kotlin please structure as`android/java/AmpliApp`

File: java/android/AmpliApp/app/src/main/java/com/amplitude/ampli/EnumCamelCase.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// [Full Setup Instructions](https://data.amplitude.com/test-codegen/Test%20Codegen/implementation/java-ampli)
+//
+package com.amplitude.ampli;
+
+import java.io.IOException;
+
+/**
+ * descriptionForEnumCamelCase
+ */
+public enum EnumCamelCase {
```
Comment (justin-fiedler) [inline java/android/AmpliApp/app/src/main/java/com/amplitude/ampli/EnumCamelCase.java:22]: Can you please namespace this like you did for Kotlin 🙏

File: java/android/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java [hunk 3]
```diff
@@ -0,0 +1,704 @@
...
+
+    public void identify(String userId, IdentifyProperties properties) {
+        this.identify(userId, properties, null, null);
+    }
+
+    public void identify(String userId, IdentifyProperties properties, EventOptions options) {
+        this.identify(userId, properties, options, null);
+    }
+
+    public void identify(String userId, IdentifyProperties properties, EventOptions options, Map<String, Object> extra) {
```
Comment (justin-fiedler) [inline java/android/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java:143]: Extra will be a JSONObject
https://github.com/amplitude/Amplitude-Android/pull/308/files#diff-f9ec4ad04ab6ea07630bc809a1f933661cb3dc4423e4d4ce07752f5128f8df0eR5

File: java/android/AmpliApp/app/src/main/java/com/example/ampliapp/App.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import com.amplitude.ampli.Ampli;
+import com.amplitude.ampli.LoadClientOptions;
+import com.amplitude.ampli.LoadOptions;
+
+public class App extends Application {
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        String amplitudeApiKey = null; // Set your Amplitude API key
```
Comment (justin-fiedler) [inline java/android/AmpliApp/app/src/main/java/com/example/ampliapp/App.java:14]: Please external (see comment in Kotlin)

**Review by justin-fiedler — CHANGES_REQUESTED** [review]
@falconandy nice work. See comments. I think the biggest improvement we could make is Builders for the Properties classes.

File: README.md (hunk 1 above)
Comment (falconandy) [reply README.md]: Done

File: java/android/AmpliApp/app/src/main/java/com/amplitude/ampli/EnumCamelCase.java (hunk 2 above)
Comment (falconandy) [reply java/android/AmpliApp/app/src/main/java/com/amplitude/ampli/EnumCamelCase.java]: Done

File: java/android/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java (hunk 3 above)
Comment (falconandy) [reply java/android/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java]: Will fix with middleware support

File: java/android/AmpliApp/app/src/main/java/com/example/ampliapp/App.java (hunk 4 above)
Comment (falconandy) [reply java/android/AmpliApp/app/src/main/java/com/example/ampliapp/App.java]: Fixed

File: java/android/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java (hunk 3 above)
Comment (falconandy) [reply java/android/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java]: Fixed

File: java/android/AmpliApp/app/src/main/java/com/example/ampliapp/App.java (hunk 4 above)
Comment (justin-fiedler) [reply java/android/AmpliApp/app/src/main/java/com/example/ampliapp/App.java]: Can you please move this to a dedicated `secrets.xml` like in Kotlin

File: android/java/AmpliApp/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Ampli Android SDK (Java)
+An example app using the Ampli Android SDK
+
+# Usage
+
+### Setup the project
+You will need to do the following before running the app.
+1. Set your Amplitude API key in `src/main/java/com/example/ampliapp/App.java` file
```
Comment (justin-fiedler) [inline android/java/AmpliApp/README.md:8]: Update to match secrets.xml resource

File: android/java/AmpliApp/README.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
```
Comment (justin-fiedler) [inline android/java/AmpliApp/README.md:9]: Please add `local.properties.example` with `#sdk.dir=/path/to/Android/sdk` and update README. See comment in Kotlin example.

File: android/java/AmpliApp/app/build.gradle [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+    implementation 'androidx.appcompat:appcompat:1.4.0'
+    implementation 'com.google.android.material:material:1.4.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.2'
+    testImplementation 'junit:junit:4.+'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+
+    implementation 'com.amplitude:android-sdk:2.35.2'
+    implementation 'com.squareup.okhttp3:okhttp:4.2.2'
```
Comment (justin-fiedler) [inline android/java/AmpliApp/app/build.gradle:40]: Update to `4.9.3` and link to github issues referenced in Kotlin example.

File: android/java/AmpliApp/app/src/androidTest/java/com/example/ampliapp/AmpliTest.java [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+
+        Plan plan = new Plan().setBranch("branch-1").setSource("source-1").setVersion("version-1");
+        this.ampli.load(appContext, new LoadOptions().setClient(new LoadClientOptions().setInstance(client).setPlan(plan)));
+
+        verify(client, times(0)).initialize(any(), any());
+        verify(client, times(1)).setPlan(plan);
+    }
+
+    @Test
+    public void identify() {
```
Comment (justin-fiedler) [inline android/java/AmpliApp/app/src/androidTest/java/com/example/ampliapp/AmpliTest.java:67]: Thanks for the tests!

**Review by justin-fiedler — CHANGES_REQUESTED** [review]

File: android/java/AmpliApp/app/build.gradle (hunk 7 above)
Comment (falconandy) [reply android/java/AmpliApp/app/build.gradle]: Updated

File: android/java/AmpliApp/README.md (hunk 6 above)
Comment (falconandy) [reply android/java/AmpliApp/README.md:24]: Added

File: android/java/AmpliApp/README.md (hunk 5 above)
Comment (falconandy) [reply android/java/AmpliApp/README.md]: Updated

File: android/java/AmpliApp/README.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    ```
+
+### Run the App
+Run the application using Android Studio or your favorite IDE.
+
+### Pull your Ampli SDK
+If you want to use your own tracking plan to generate the Ampli SDK:
+1. `npm i -g @amplitude/ampli`
+2. `ampli pull`
+3. Update [MainActivity.java](app/src/main/java/com/example/ampliapp/MainActivity.java) to track your events.
```
Comment (justin-fiedler) [inline android/java/AmpliApp/README.md:29]: Thanks for updating to latest from Kotlin, and fixing the relative paths.

**Review by justin-fiedler — APPROVED** [review]

**Pull request merged**